asyncpraw/models/reddit/subreddit.py

```diff
@@ -647,21 +676,23 @@ def _upload_media(self, media_path, expected_mime_prefix=None):
 
         url = API_PATH["media_asset"]
         # until we learn otherwise, assume this request always succeeds
-        upload_lease = self._reddit.post(url, data=img_data)["args"]
+        response = await self._reddit.post(url, data=img_data)
+        upload_lease = response["args"]
         upload_url = "https:{}".format(upload_lease["action"])
         upload_data = {item["name"]: item["value"] for item in upload_lease["fields"]}
 
         with open(media_path, "rb") as media:
-            response = self._reddit._core._requestor._http.post(
-                upload_url, data=upload_data, files={"file": media}
+            upload_data["file"] = media
+            response = await self._reddit._core._requestor._http.post(
+                upload_url, data=upload_data
             )
-        if not response.ok:
```

Comment:
200 was ok in requests
https://github.com/psf/requests/blob/master/requests/status_codes.py

But now it has to be 201?

Reply:
aiohttp does not have an `.ok` attr on its responses.

Reply:
The `response.ok` actually corresponds with this:
https://github.com/psf/requests/blob/1b417634721ec377abb7f17bc1f215e07202c2f7/requests/models.py#L695-L708

Reply:
@LilSpazJoekp 
Sorry, I was not clear in my message.

the `response.ok` in praw has been changed to `response.status==201`
So the 200 has been changed to 201. I am not sure if it comes with the modification of requests to aiohttp. Or if it is a bug that needs to be fixed

Reply:
I understood what you meant. I'm saying that `response.ok` in this context does not mean status is 200. According to the requests docs I linked above, it will return `True` if the status is <400. Since `aiohttp` does not have this method on its requests I switched it to check if it `201`, the code it returns on a successful request.